opennms-alarms/daemon/src/test/java/org/opennms/netmgt/alarmd/AlarmdIT.java

```diff
@@ -278,48 +283,58 @@ public void run() {
-        assertEquals(1, rowCount);
-        if (numberOfAlarmsToReduce != counterColumn) {
-            final List<Integer> reducedEvents = new ArrayList<>();
-            m_jdbcTemplate.query("select eventid from events where alarmID is not null", new RowCallbackHandler() {
-                @Override
-                public void processRow(ResultSet rs) throws SQLException {
-                    reducedEvents.add(rs.getInt(1));
-                }
-            });
-            Collections.sort(reducedEvents);
+        int alarmCount = m_alarmDao.findAll().size();
+        assertEquals("there should be 1 alarm", 1, alarmCount);
+        OnmsAlarm alarm = m_alarmDao.findByReductionKey(reductionKey);
+        Integer counter = alarm.getCounter();
+        MockUtil.println("rowcCount is: "+alarmCount+", expected 1.");
+        MockUtil.println("counterColumn is: "+counter+", expected "+numberOfAlarmsToReduce);
+        if (numberOfAlarmsToReduce != counter) {
+            fail("number of alarms to reduce (" + numberOfAlarmsToReduce + ") were not reduced into a single alarm (instead the counter column reads " + counter + "); ");
+        }
+    }
 
-            final List<Integer> nonReducedEvents = new ArrayList<>();
-            m_jdbcTemplate.query("select eventid from events where alarmID is null", new RowCallbackHandler() {
-                @Override
-                public void processRow(ResultSet rs) throws SQLException {
-                    nonReducedEvents.add(rs.getInt(1));
-                }
-            });
-            Collections.sort(nonReducedEvents);
+    @Test
+    @Transactional
+    public void testPersistSituations() throws Exception {
+        final MockNode node = m_mockNetwork.getNode(1);
 
-            fail("number of alarms to reduce (" + numberOfAlarmsToReduce + ") were not reduced into a single alarm (instead the counter column reads " + counterColumn + "); "
-                    + "events that were reduced: " + StringUtils.collectionToCommaDelimitedString(reducedEvents) + "; events that were not reduced: "
-                    + StringUtils.collectionToCommaDelimitedString(nonReducedEvents));
-        }
+        //there should be no alarms in the alarms table
+        assertEmptyAlarmTable();
 
+        //there should be no alarms in the alarm_situations table
+        assertEmptyAlarmSituationTable();
 
-        Integer alarmId = m_jdbcTemplate.queryForObject("select alarmId from alarms where reductionKey = ?", new Object[] { reductionKey }, Integer.class).intValue();
-        rowCount = m_jdbcTemplate.queryForObject("select count(*) from events where alarmid = ?", new Object[] { alarmId }, Integer.class).intValue();
-        MockUtil.println(String.valueOf(rowCount) + " of events with alarmid: "+alarmId);
-        //      assertEquals(numberOfAlarmsToReduce, rowCount);
+        //create 3 alarms to roll up into situation
+        sendNodeDownEvent("Alarm1", node);
+        Thread.sleep(1000);
```

Comment:
I'd opt to use `await().until()` for checking asynchronous updates instead of relying to fixed sleep times in this new test. 1) tests tend to finish faster and 2) they work more reliably